src/session/session_dhandle.c

```diff
@@ -473,20 +474,29 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
      * confusion caused by older versions that don't include these values.
      *
      * Also note that only the exact name "WiredTigerCheckpoint" needs to be resolved. Requests to
-     * open specific versions, such as "WiredTigerCheckpoint.6", must be looked up like named
-     * checkpoints but are otherwise still treated as unnamed. This is necessary so that the
-     * matching history store checkpoint we find can be itself opened later.
+     * open specific versions, such as "WiredTigerCheckpoint.6", is forbidden.
      *
      * It is also at least theoretically possible for there to be no matching history store
      * checkpoint. If looking up the checkpoint names finds no history store checkpoint, its name
      * will come back as null and we must avoid trying to open it, either here or later on in the
      * life of the checkpoint cursor.
      */
 
-    if (strcmp(uri, WT_HS_URI) == 0)
+    is_hs = strcmp(uri, WT_HS_URI) == 0;
+    if (is_hs)
         /* We're opening the history store directly, so don't open it twice. */
         hs_dhandlep = NULL;
 
+    /*
+     * The internal checkpoint name is special, applications aren't allowed to use it. Be aggressive
+     * and disallow any matching prefix. However, internally we may want to open the history store
+     * with a specific version, this is allowed.
+     */
+    is_reserved_name = cval.len > strlen(WT_CHECKPOINT) && WT_PREFIX_MATCH(cval.str, WT_CHECKPOINT);
```

Comment:
I wonder if it should be `>=` of the strlen, should we disallow exactly `WiredTigerCheckpoint` as well? Or am I off by one and it already works that way?

Reply:
@etienneptl politely pointed out that the decision is deliberate. Applications wanting to open a default checkpoint need to pass in the prefix here.

Could you please update the comment to make that more clear?

Reply:
@clarissecheah updated the comment in this commit: e75a1b30e8d2aa07ce616459173ea24a3b3ebd8f. Thanks!